node/src/main/scala/coop/rchain/node/web/WebApiDocsV1.scala

```diff
@@ -0,0 +1,47 @@
+package coop.rchain.node.web
+
+import cats.effect.Sync
+import coop.rchain.node.api.v1.{WebApiAdminEndpoints, WebApiEndpoints}
+import endpoints4s.http4s.server
+import endpoints4s.http4s.server.Endpoints
+import endpoints4s.openapi
+import endpoints4s.openapi.model.{Info, OpenApi}
+
+import scala.Function.const
+
+/**
+  * OpenAPI schema definition for RNode Web API v1.
+  */
+object WebApiDocs
+    extends WebApiEndpoints
+    with WebApiAdminEndpoints
+    with openapi.Endpoints
+    with openapi.JsonEntitiesFromSchemas {
+
+  val public =
+    Seq(status, deploy, exploreDeploy, exploreDeployByBlockHash, dataAtName, getBlocks, getBlock)
+
+  val admin = Seq(propose)
+
+  // Public API Open API schema
+  val publicApi: OpenApi = openApi(Info(title = "RNode API", version = "1.0"))(public: _*)
+
+  // Admin API includes the whole public API schema
+  val adminApi: OpenApi =
+    openApi(Info(title = "RNode API (admin)", version = "1.0"))(public ++ admin: _*)
+}
+
+/**
+  * OpenAPI endpoint definition (GET /openapi.json).
+  */
+final case class WebApiDocServer[F[_]: Sync]()
+    extends Endpoints[F]
+    with server.JsonEntitiesFromEncodersAndDecoders {
+  implicit val jCodec: endpoints4s.Encoder[OpenApi, String] = OpenApi.stringEncoder
+
+  val publicRoutes = endpoint(get(path / "openapi.json"), ok(jsonResponse[OpenApi]))
+    .implementedBy(const(WebApiDocs.publicApi))
```

Comment:
just curious about this `const`. Is it just marker or it can be used by compiler to optimize?

Reply:
`Function.const` is just a built-in helper function to ignore the first argument. It's the same as:
```scala
implementedBy(_ => WebApiDocs.publicApi)
```
I'm not aware that compiler is doing some optimization.